app.rb

```diff
@@ -0,0 +1,160 @@
+require_relative 'person'
+require_relative 'book'
+require_relative 'rental'
+require_relative 'student'
+require_relative 'teacher'
+
+class App
+  def initialize
+    @people = [] # Array to store people (students/teachers)
+    @books = [] # Array to store books
+    @rentals = [] # Array to store rentals
+  end
+
+  def list_all_books
+    puts 'List of all books:'
+    @books.each do |book|
+      puts "Title: #{book.title}, Author: #{book.author}"
+    end
+  end
+
+  def list_all_people
+    puts 'List of all people:'
+    @people.each do |person|
+      puts "ID: #{person.id}, Name: #{person.name}, Age: #{person.age}"
+    end
+  end
+
+  def create_person(role)
+    puts "Creating a new #{role}..."
+    puts 'Enter name:'
+    name = gets.chomp
+    puts 'Enter age:'
+    age = gets.chomp.to_i
+  
+    case role
+    when 'student'
+      if age < 18
+        puts 'Does the student have parent permission? (yes/no):'
+        permission_input = gets.chomp.downcase
+        parent_permission = permission_input == 'yes'
+      else
+        parent_permission = true
+      end
+  
+      person = Student.new(age, name, parent_permission: parent_permission)
+    when 'teacher'
+      puts 'Enter specialization:'
+      specialization = gets.chomp
+      person = Teacher.new(specialization, name: name, age: age)
+    end
+  
+    @people << person
+  end
+
+  def create_book
+    puts 'Creating a new book...'
+    puts 'Enter Author:'
+    title = gets.chomp
+    puts 'Enter Title:'
+    author = gets.chomp
+    book = Book.new(author, title)
+    @books << book
+  end
+
+  def create_rental
```

Comment:
Incredible work with the Ruby Basic UI application. 👏🏾 

Kindly note that the `create rental` functionality in the [example video](https://www.youtube.com/watch?v=vkkgrhD6aXQ) should be used as a guideline for the Basic UI implementation.

- Your implementation requires the user to input the person ID and book title manually. The application should use the list of people and books from the variables as options to choose from. This will ensure consistent data entry in the application.

![image](https://github.com/Ochiengsteven/school-library-oop/assets/38088818/766ffd73-2630-4dc8-857e-b01c07582e72)